extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -781,10 +783,22 @@ private static HttpServerOptions createSslOptions(HttpBuildTimeConfig buildTimeC
             certificates.add(certFile.get());
         }
 
+        // credentials provider
+        Map<String, String> credentials = Map.of();
+        if (sslConfig.certificate.credentialsProvider.isPresent()) {
+            String beanName = sslConfig.certificate.credentialsProviderName.orElse(null);
+            CredentialsProvider credentialsProvider = CredentialsProviderFinder.find(beanName);
```

Comment:
How does this lookup work? Is it using `@Identifier`?

Reply:
Yeah... it uses @Named. I don't believe we want this. @mkouba WDYT? 
My guess is that a proper qualifier would be better.

Reply:
@cescoffier Sure; `@Named` would only come into effect if there are more than one `CredentialsProvider` loaded, with a single one it make no difference. Not sure we can change that though as a few such providers use `@Named` (the one in `quarkus-vault` for example)

Reply:
@cescoffier Sure; `@Named` would only come into effect if there are more than one `CredentialsProvider` loaded, with a single one it make no difference. Not sure we can change that though as a few such providers use `@Named` (the one in `quarkus-vault` for example)

Reply:
> Yeah... it uses @nAmed. I don't believe we want this. @mkouba WDYT? My guess is that a proper qualifier would be better.

+1

Reply:
I guess it is not something that we can do in this PR, right ?

Reply:
@cescoffier @mkouba As far as this PR is concerned, I can make a `credentials-provider-name` property not optional but required if you are concerned about a random provider being picked up, it will ensure that only a provider with a matching `Named` value will be picked up. 

I think the actual `CredentialsProvider` resolution logic update is out of scope for this PR as many extensions depend on it (making the `credentials-provider-name` required for all those extensions can be the 1st step forward), changing it willbe a breaking change and will make this PR non-backportable if it will have to be backported.

Reply:
@cescoffier I've noticed you've approved, thanks, but your concern about `@Named` is noted for sure. As I said, I can enforce at the vertx-http level that `credentials-provider-name` is required, however, I'm having a 2nd thought about it now, if `CredentialsProvider` resolution logic will indeed get changed in the future then having `credentials-provider-name` explicitly required may make it harder to migrate.
Perhaps a reasonable compromise is to recommend in JavaDocs of this property that it should be used, will do it now

Reply:
Just pushed a JavaDoc update recommending to use the name property all the time